### ISSUE TITLE
[<Ray component: Core|RLlib|etc...>] 

### ISSUE THREAD

**Issue by amadou1998** [ray-project/ray#32467]
### Description

Time attr is described by:

```
"""        
time_attr: A training result attr to use for comparing time.
            Note that you can pass in something non-temporal such as
            `training_iteration` as a measure of progress, the only 
"""
```
Something like such as is kind of very unsepcific. What is meant by that? What can i pass it without the run erroring out? I tried some stuff but would it not be better having an exhaustive list of options from which to choose? I was unable to find a better description of this anywhere. In the stoppers it says we can also pass a dict, but here aswell, the keys of the dict are not specified. If I pass any key it crashes.

**Comment by xwjiang2010** [comment]
Can you provide your script? It's helpful for me to understand how to improve the documentation here by looking at some misconceptions.

**Comment by amadou1998** [comment]
I am unable to provide a repro script as I am working in company. However this does not need any software as it is simply not described in the documentation, what other option than "training_iteration" exist by default in ray. This does not depend whatever code I write or use this with.

**Comment by justinvyu** [comment]
Hi @amadou1998,

`time_total_s` and `training_iteration` are the common monotonically increasing metrics that you use with schedulers. You're right, we should include the exact list of metrics to choose from, out of the ones Tune provides automatically.

The reason why we can't have an exhaustive list is that this could also use a metric that you report -- for example, you may track the number of training steps.

**Comment by anthropikos** [comment]
I am also a bit confused on this parameter of the scheduler. 

It seems that `time_attr` can take any metric that is being reported to the Trainer, is that correct? 

Additionally, regarding the default value of `training_iteration`, does it refer to `epoch` as the step metric or the `step` of either `train_dataloader` or `val_dataloader`?

Feel free to ask clarification questions because I have attempted to understand this by viewing the source code but my understanding of how RayTune, RayTrain, and others work is still fuzzy.

For example, here is the output of the `lightning.pytorch.loggers.CSVLogger` from one epoch of training.
I think the `step` column here is the index of the last item of each batch iterated by the `train_dataloader`

```csv
epoch,step,train_loss_epoch,train_loss_step,val_loss
0,49,,0.6689074635505676,
0,99,,0.6776237487792969,
0,149,,0.6492859125137329,
0,199,,0.638532280921936,
0,249,,0.6305821537971497,
0,299,,0.6167992949485779,
0,349,,0.6237543225288391,
0,399,,0.6124219298362732,
0,449,,0.5973061323165894,
0,499,,0.5894381999969482,
0,549,,0.6079745888710022,
0,599,,0.6158718466758728,
0,649,,0.5484498739242554,
0,699,,0.5869404673576355,
0,749,,0.6183856725692749,
0,799,,0.5806534886360168,
0,849,,0.5823719501495361,
0,899,,0.5955649614334106,
0,949,,0.5673904418945312,
0,999,,0.591978907585144,
0,1049,,0.5745832324028015,
0,1099,,0.5654190182685852,
0,1149,,0.5597402453422546,
0,1199,,0.5518961548805237,
0,1249,,0.5515714287757874,
0,1299,,0.5010809898376465,
0,1349,,0.48797932267189026,
0,1399,,0.5497390031814575,
0,1449,,0.4611366093158722,
0,1499,,0.43229541182518005,
0,1549,,0.4488227069377899,
0,1599,,0.43265628814697266,
0,1649,,0.38390228152275085,
0,1699,,0.34099337458610535,
0,1749,,0.39753270149230957,
0,1799,,0.39656656980514526,
0,1849,,0.38244500756263733,
0,1899,,0.36464986205101013,
0,1949,,0.3146722614765167,
0,1999,,0.33978551626205444,
0,2049,,0.3168078064918518,
0,2099,,0.3053840696811676,
0,2149,,0.3293323218822479,
0,2199,,0.3171581029891968,
0,2249,,0.2852384150028229,
0,2299,,0.28903865814208984,
0,2349,,0.2889100909233093,
0,2399,,0.25731146335601807,
0,2449,,0.22585193812847137,
0,2499,,0.24324730038642883,
0,2549,,0.27017274498939514,
0,2599,,0.24094092845916748,
0,2649,,0.23172636330127716,
0,2699,,0.20534424483776093,
0,2749,,0.2802880108356476,
0,2799,,0.20207561552524567,
0,2849,,0.21664221584796906,
0,2899,,0.2572675347328186,
0,2949,,0.18741294741630554,
0,2999,,0.17474137246608734,
0,3049,,0.18670505285263062,
0,3099,,0.17625059187412262,
0,3149,,0.15423709154129028,
0,3199,,0.13702456653118134,
0,3249,,0.1823478490114212,
0,3299,,0.16864417493343353,
0,3349,,0.13686875998973846,
0,3399,,0.1171843484044075,
0,3449,,0.17752160131931305,
0,3499,,0.15154653787612915,
0,3549,,0.14218354225158691,
0,3599,,0.15925197303295135,
0,3649,,0.12001139670610428,
0,3699,,0.15289999544620514,
0,3749,,0.19480809569358826,
0,3799,,0.13536497950553894,
0,3849,,0.11786238849163055,
0,3899,,0.12974189221858978,
0,3949,,0.14271140098571777,
0,3999,,0.16013841331005096,
0,4049,,0.12746666371822357,
0,4099,,0.13505183160305023,
0,4149,,0.12548165023326874,
0,4199,,0.12722168862819672,
0,4249,,0.13262632489204407,
0,4299,,0.13541902601718903,
0,4349,,0.12377989292144775,
0,4399,,0.11905107647180557,
0,4449,,0.12075798213481903,
0,4499,,0.15814658999443054,
0,4549,,0.13469132781028748,
0,4599,,0.14071601629257202,
0,4649,,0.12705808877944946,
0,4699,,0.15195606648921967,
0,4749,,0.1694248765707016,
0,4799,,0.11281488835811615,
0,4849,,0.09977320581674576,
0,4899,,0.1038784608244896,
0,4949,,0.12216147780418396,
0,4999,,0.11964709311723709,
0,5049,,0.11980345100164413,
0,5099,,0.09321320801973343,
0,5149,,0.11077607423067093,
0,5199,,0.11326512694358826,
0,5249,,0.12040987610816956,
0,5299,,0.10987313091754913,
0,5349,,0.1275690197944641,
0,5399,,0.1333008110523224,
0,5449,,0.11249875277280807,
0,5499,,0.10821262001991272,
0,5549,,0.11086280643939972,
0,5599,,0.10991518944501877,
0,5649,,0.10894276946783066,
0,5699,,0.12447398155927658,
0,5749,,0.092387855052948,
0,5799,,0.10403723269701004,
0,5849,,0.12323439866304398,
0,5899,,0.11648282408714294,
0,5949,,0.11427934467792511,
0,5999,,0.10274333506822586,
0,6049,,0.10248137265443802,
0,6099,,0.0944165363907814,
0,6149,,0.11312215030193329,
0,6199,,0.11783217638731003,
0,6249,,0.12311185896396637,
0,6299,,0.10140267014503479,
0,6349,,0.14301592111587524,
0,6399,,0.11471442878246307,
0,6449,,0.11229504644870758,
0,6499,,0.10951095819473267,
0,6549,,0.11862701922655106,
0,6599,,0.1271309107542038,
0,6649,,0.09453373402357101,
0,6699,,0.09039365500211716,
0,6749,,0.10284784436225891,
0,6799,,0.13178220391273499,
0,6849,,0.1329018622636795,
0,6899,,0.09629040211439133,
0,6949,,0.12371056526899338,
0,6999,,0.10492783039808273,
0,7049,,0.10798242688179016,
0,7099,,0.12651708722114563,
0,7149,,0.12470901757478714,
0,7199,,0.10175783187150955,
0,7249,,0.12412726134061813,
0,7299,,0.12168800085783005,
0,7349,,0.10165289789438248,
0,7399,,0.11284862458705902,
0,7449,,0.10079415142536163,
0,7499,,0.10253589600324631,
0,7549,,0.11769271641969681,
0,7599,,0.1358717381954193,
0,7649,,0.1019330844283104,
0,7699,,0.1094466969370842,
0,7749,,0.1384751945734024,
0,7799,,0.1318492293357849,
0,7849,,0.10680697858333588,
0,7899,,0.10206831991672516,
0,7949,,0.09622573107481003,
0,7999,,0.11157877743244171,
0,8049,,0.0887809619307518,
0,8099,,0.11150962114334106,
0,8149,,0.10443874448537827,
0,8199,,0.10518188774585724,
0,8249,,0.09769972413778305,
0,8299,,0.11307491362094879,
0,8349,,0.12959031760692596,
0,8399,,0.10731863230466843,
0,8449,,0.08946345746517181,
0,8499,,0.09566117078065872,
0,8549,,0.08657150715589523,
0,8599,,0.0866052657365799,
0,8649,,0.10247509926557541,
0,8699,,0.08528981357812881,
0,8749,,0.11387574672698975,
0,8799,,0.0881294459104538,
0,8849,,0.08635910600423813,
0,8899,,0.10868694633245468,
0,8949,,0.1263614445924759,
0,8999,,0.07982748746871948,
0,9049,,0.10863666236400604,
0,9099,,0.10305118560791016,
0,9149,,0.2262813150882721,
0,9199,,0.12901313602924347,
0,9249,,0.11768218874931335,
0,9299,,0.11651092022657394,
0,9349,,0.10845949500799179,
0,9399,,0.13197734951972961,
0,9449,,0.10590773820877075,
0,9499,,0.11612067371606827,
0,9549,,0.10344396531581879,
0,9599,,0.09643801301717758,
0,9649,,0.09489931911230087,   # Here and prior marks `train_loss` at each step of `train_dataloader`
0,9698,,,0.18093079328536987  # `val_loss`
0,9698,0.21968774497509003,,  # `train_loss_epoch`
```
